### PR TITLE
Links: Fix opening links from different orgs on the same tab

### LINKED ISSUE #55326: Linking to a dashboard of a different org only works on a new tab
**What happened**:

When I create a link to a dashboard passing the orgId on the URL, even if I have access to such org, when I click the link and open it on the same tab I get a "Dashboard not found" message, but if I open the link on a new tab I don't have any issues.

**What you expected to happen**:

I expected to be able to link to dashboards of different orgs than the current one if I have access to those orgs and open the dashboards on the same tab, without requiring a new tab.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a dashboard on the main org
2. Create a new org
3. Create a dashboard on the new org
4. Add a text panel to the dashboard on the new org with a link to the dashboard on the main org
5. Click the link, it will fail to open on the same tab
6. Click the link with the hotkey to open a new tab pressed (most times it's CTRL), it works

**Anything else we need to know?**:

**Environment**:
- Grafana version: main(https://github.com/grafana/grafana/commit/5c66c412200b8f756851ef84da0a4130b9bbc16a)
- Data source type & version: Prometheus (prom/prometheus:latest) from devenv
- OS Grafana is installed on: macOS Monterey 12.6
- User OS & Browser: Chrome Version 105.0.5195.125 (Official Build) (arm64) on macOS Monterey 12.6
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by guicaulada**
**What this PR does / why we need it**:

Currently when linking to a different Org we do a single page transition, resulting in the grafanaBootData.user.orgId being different than the one passed on the URL, the same happens for the user session on the backend. With this change we will do a full reload when linking to different orgs, this way the orgId in the URL, grafanaBootData and backend will be in sync.

This will solve issues with linking to dashboads of different orgs on Text Panels and Dashboard Links.
Currently those links only work if opened on a new tab, this will make them work opening on the same tab.
Having the links work only on a new tab is specially bad on the Text Panel when using Markdown since it's impossible to specify that the link should open on a new tab.

**Which issue(s) this PR fixes**:

Fixes #55326

**Special notes for your reviewer**:

### CODE REVIEW

**Review by owensmallwood — APPROVED** [review]
Works as expected 👍

File: public/app/core/navigation/patch/interceptLinkClicks.ts [hunk 1]
```diff
@@ -14,6 +15,8 @@ export function interceptLinkClicks(e: MouseEvent) {
     const target = anchor.getAttribute('target');
 
     if (href && !target) {
+      const params = urlUtil.parseKeyValue(href.split('?')[1]);
```
Comment (juanicabanas) [inline public/app/core/navigation/patch/interceptLinkClicks.ts:18]: there's a locationSearchToObject method in LocationService. Maybe we can reuse it

**Review by juanicabanas — APPROVED** [review]

File: public/app/core/navigation/patch/interceptLinkClicks.ts (hunk 1 above)
Comment (guicaulada) [reply public/app/core/navigation/patch/interceptLinkClicks.ts]: Done on https://github.com/grafana/grafana/pull/55837/commits/f1f6c6c29ea1a988b96fa04a1b4a57a0a0e97c75

Comment (guicaulada) [reply public/app/core/navigation/patch/interceptLinkClicks.ts:18]: Reverted on https://github.com/grafana/grafana/pull/55837/commits/557a3c7b5770dc788d8fbcdb166a33a30c7780ea due to this function locationSearchToObject not dealing well with undefined values.

**Pull request merged**